drain-service/drain_training_inferencing.py

```diff
@@ -359,6 +360,23 @@ async def init_nats():
     logging.info("connecting to nats")
     await nw.connect()
 
+async def wait_for_index():
```

Comment:
question here: 
1. Why we need this piece of code now?
2.  should we make it as the script of  readinessProbe of this service?

Reply:
The service is waiting for Elasticsearch to get ready, but as part of the reconciliation loop in the operator we are setting up the opni-normal-intervals index with mappings.  We want to ensure that we're not indexing any docs to the index without the mappings.

With regards to the readiness probes, in general that's a good idea, however a pod that's in a not ready state only gets removed from the kubernetes Service endpoints.  This won't stop it consuming from the NATs subject and working through its logic

Reply:
This makes sense. but then why wouldn't the nulog-inference service need the same logic then?

Reply:
In regards to which indices?  My understanding is that the inference service will only update indices if it retrieves documents.